GitUI/CommandsDialogs/FormPush.Designer.cs

```diff
@@ -207,6 +207,7 @@ private void InitializeComponent()
             _NO_TRANSLATE_Branch.TabIndex = 0;
             _NO_TRANSLATE_Branch.SelectedIndexChanged += _NO_TRANSLATE_Branch_SelectedIndexChanged;
             _NO_TRANSLATE_Branch.SelectedValueChanged += BranchSelectedValueChanged;
+            _NO_TRANSLATE_Branch.Enter += _NO_TRANSLATE_Branch_Enter;
```

Comment:
nit pick
Why not DropDownOpening?
Sure a few ms saved for the user but will run more often

Reply:
So that it doesn't break the keyboard workflow.

>Sure a few ms saved for the user but will run more often

It could save a lot more depending how the user interact with it. And perf is not really the issue as there is a condition that exits early if data are already loaded.